genesyscloud/resource_genesyscloud_outbound_callanalysisresponseset.go

```diff
@@ -0,0 +1,386 @@
+				Type:     schema.TypeSet,
+				MaxItems: 1,
+				Optional: true,
+				Elem:     outboundCallAnalysisResponseSetReaction,
+			},
+			`callable_machine`: {
+				Computed: true,
+				Type:     schema.TypeSet,
+				MaxItems: 1,
+				Optional: true,
+				Elem:     outboundCallAnalysisResponseSetReaction,
+			},
+			`callable_sit`: {
+				Computed: true,
+				Type:     schema.TypeSet,
+				MaxItems: 1,
+				Optional: true,
+				Elem:     outboundCallAnalysisResponseSetReaction,
+			},
+			`uncallable_sit`: {
+				Computed: true,
+				Type:     schema.TypeSet,
+				MaxItems: 1,
+				Optional: true,
+				Elem:     outboundCallAnalysisResponseSetReaction,
+			},
+			`uncallable_notfound`: {
+				Computed: true,
+				Type:     schema.TypeSet,
+				MaxItems: 1,
+				Optional: true,
+				Elem:     outboundCallAnalysisResponseSetReaction,
+			},
+		},
+	}
+)
+
+func getAllCallAnalysisResponseSets(_ context.Context, clientConfig *platformclientv2.Configuration) (ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(ResourceIDMetaMap)
+	outboundAPI := platformclientv2.NewOutboundApiWithConfig(clientConfig)
+
+	for pageNum := 1; ; pageNum++ {
+		const pageSize = 100
+		responseSetConfigs, _, getErr := outboundAPI.GetOutboundCallanalysisresponsesets(pageSize, pageNum, true, "", "", "", "")
+		if getErr != nil {
+			return nil, diag.Errorf("Failed to get page of call analysis response set configs: %v", getErr)
+		}
+		if responseSetConfigs.Entities == nil || len(*responseSetConfigs.Entities) == 0 {
+			break
+		}
+		for _, responseSetConfig := range *responseSetConfigs.Entities {
+			resources[*responseSetConfig.Id] = &ResourceMeta{Name: *responseSetConfig.Name}
+		}
+	}
+
+	return resources, nil
+}
+
+func outboundCallAnalysisResponseSetExporter() *ResourceExporter {
+	return &ResourceExporter{
+		GetResourcesFunc: getAllWithPooledClient(getAllCallAnalysisResponseSets),
+		RefAttrs: map[string]*RefAttrSettings{
+			"responses.callable_person.data":  {RefType: "genesyscloud_flow"},
+			"responses.callable_machine.data": {RefType: "genesyscloud_flow"},
+		},
+	}
+}
+
+func resourceOutboundCallAnalysisResponseSet() *schema.Resource {
+	return &schema.Resource{
+		Description: `Genesys Cloud outbound Call Analysis Response Set`,
+
+		CreateContext: createWithPooledClient(createOutboundCallAnalysisResponseSet),
+		ReadContext:   readWithPooledClient(readOutboundCallAnalysisResponseSet),
+		UpdateContext: updateWithPooledClient(updateOutboundCallAnalysisResponseSet),
+		DeleteContext: deleteWithPooledClient(deleteOutboundCallAnalysisResponseSet),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			`name`: {
+				Description: `The name of the Response Set.`,
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			`responses`: {
+				Description: `Map of disposition identifiers to reactions. Required if beep_detection_enabled = true.`,
```

Comment:
Description should be updated since it's not a map

Reply:
Maybe it should be `List of maps of disposition ...`

Reply:
@ronanwatkins Done ✅